src/Route.php

```diff
@@ -208,6 +208,9 @@ public function getOptions()
     /**
      * Whether or not HEAD support is implicit (i.e., not explicitly specified)
      *
+     * @deprecated Since 2.4.0; to be removed in 3.0.0. Router implementations
```

Comment:
What about trigger_error deprecated in these deprecated methods?
It is much easier updating code when you can see deprecated notices (let say on running unit test of your code) than just based on annotation in code.

Reply:
Can't, because _our own middleware uses them_. We'd have the notices triggered for every single HEAD and OPTIONS request!